### PR TITLE
touchdetector, morphio: update version, dependencies.

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: var/spack/repos/builtin/packages/morphio/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Morphio(CMakePackage):
+    """Library for reading / writing morphology files"""
+
+    homepage = "https://github.com/BlueBrain/MorphIO"
+    url      = "git@github.com:BlueBrain/MorphIO.git"
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/morphio/package.py]: url with https ?

File: var/spack/repos/builtin/packages/morphio/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    version('develop', git=url, submodules=True)
+    version('2.0.8', tag='v2.0.8', git=url, submodules=True, preferred=True)
+
+    depends_on('cmake@3.2:', type='build')
+    depends_on('hdf5')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/morphio/package.py:19]: minor : As hdf5 by default has +mpi, I wonder if we should explicitly start adding ~mpi.

File: var/spack/repos/builtin/packages/touchdetector/package.py [hunk 3]
```diff
@@ -38,16 +38,20 @@ class Touchdetector(CMakePackage):
...
-    depends_on('boost@1.50:')
     depends_on('cmake', type='build')
+    depends_on('boost@1.50:')
+    depends_on('hdf5@1.8:')
     depends_on('hpctools~openmp', when='~openmp')
     depends_on('hpctools+openmp', when='+openmp')
-    depends_on('hdf5@1.8:')
-    depends_on('libxml2')
+    depends_on('morphio@2.0.8:', when='@4.5:')
+    depends_on('mvdtool@1.5.1:', when='@4.5:')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/touchdetector/package.py:47]: minor : there is no `4.5` version

**Review by pramodk — APPROVED** [review]

File: var/spack/repos/builtin/packages/touchdetector/package.py (hunk 3 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/touchdetector/package.py:47]: Yes, I know :)

This is for the future!

File: var/spack/repos/builtin/packages/morphio/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/morphio/package.py]: Same as for `py-morphio`.

Comment (pramodk) [reply var/spack/repos/builtin/packages/morphio/package.py]: I didnt get you....MorphIO is public now..so we should change both then

**Comment by arsenius7** [comment]
@matz-e , somehow modules dependent on `py-morphio` are not happy with this change, and I have no clue where to look for the actual reason.

```
source /gpfs/bbp.cscs.ch/apps/hpc/jenkins/config/modules.sh
module show placement-algorithm 2>&1 | grep morphio
```
now gives me:
```
prepend-path PYTHONPATH /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/applications/2018-12-19/linux-rhel7-x86_64/gcc-6.4.0/py-morphio-2.0.8-iglo5jerdk/lib/python3.6/site-packages
```
but such a path does not exists.

**Pull request merged**